compiler/crates/relay-typegen/src/visit.rs

```diff
@@ -588,6 +625,35 @@ fn import_relay_resolver_function_type(
         &PathBuf::from(resolver_metadata.import_path.lookup()),
     );
 
+    let context_import = if resolver_metadata.live {
+        match &typegen_context
+            .project_config
+            .typegen_config
+            .live_resolver_context_type
+        {
+            Some(LiveResolverContextTypeInput::Path(context_import)) => {
+                Some(LiveResolverContextType {
+                    name: context_import.name.clone().intern(),
+                    import_path: typegen_context.project_config.js_module_import_identifier(
+                        &typegen_context.project_config.artifact_path_for_definition(
+                            typegen_context.definition_source_location,
+                        ),
+                        &PathBuf::from(&context_import.path),
+                    ),
+                })
+            }
+            Some(LiveResolverContextTypeInput::Package(context_import)) => {
+                Some(LiveResolverContextType {
+                    name: context_import.name.clone().intern(),
+                    import_path: context_import.package.clone().intern(),
+                })
+            }
+            None => None,
+        }
+    } else {
+        None
+    };
```

Comment:
Since only the `import_path` is different we could do something like this (Note I didn't run this)

I think it doesn't matter too much but it might make it a bit clearer.

```suggestion
    let context_import = if resolver_metadata.live {
        match &typegen_context
            .project_config
            .typegen_config
            .live_resolver_context_type
        {
            Some(context_type) => {
                let import_path = match context_type {
                    LiveResolverContextTypeInput::Path(context_import) => {
                        typegen_context.project_config.js_module_import_identifier(
                            &typegen_context.project_config.artifact_path_for_definition(
                                typegen_context.definition_source_location,
                            ),
                            &PathBuf::from(&context_import.path),
                        )
                    }
                    LiveResolverContextTypeInput::Package(context_import) => {
                        context_import.package.clone().intern()
                    }
                };

                Some(LiveResolverContextType {
                    name: context_type.name.clone().intern(),
                    import_path,
                })
            },
            None => None,
        }
    } else {
        None
    };
```

Reply:
I don't think this work because in the `Some(LiveResolverContextType {` I get an error `no field `name` on type `&LiveResolverContextTypeInput` because at that point the type isn't specific enough to know that both types share the `name` field